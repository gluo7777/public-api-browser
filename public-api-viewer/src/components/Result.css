#results ul{
    height: 80vh;
    overflow-y: scroll;
}