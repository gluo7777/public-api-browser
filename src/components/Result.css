#results ul{
    height: 80vh;
    overflow-y: scroll;
}

.highlight {
    background-color: yellow;
}