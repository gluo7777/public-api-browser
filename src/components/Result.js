import './Result.css'
import React from 'react'

const ResultList = (props) => (
    <div id="results" className="card">
        <ul className="list-group list-group-flush">
            {props.results ?
                props.results.map((result, i) => <Result key={i} text={result['API']} />)
                : null}
        </ul>
    </div>
);

// <strong>API #1</strong> Cras <span className="highlight">justo</span> odio

const Result = (props) => (
    <li className="list-group-item">{props.text}</li>
);

export default ResultList;