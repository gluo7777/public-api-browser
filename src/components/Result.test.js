import { Result } from './Result'
import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import { ReactTestUtils, act } from 'react-dom/test-utils'

// snapshot testing
test('Result renders columns correctly', () => {
    const comp = renderer.create(<Result onRowClicked={Function()}>
        <p><span>column1</span></p>
        <p><span>column2</span></p>
    </Result>);
    expect(comp.toJSON()).toMatchSnapshot();
});

// dom testing (simulates real browser usage)
let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

it('Result renders correctly in DOM and click triggers callback', () => {
    const onClick = jest.fn();
    act(() => {
        ReactDOM.render(<Result onRowClicked={onClick}>
            <p><span>column1</span></p>
            <p><span>column2</span></p>
        </Result>, container);
    });

    // test tree
    const row = container.querySelector('.result');
    expect(row).not.toBeNull();
    expect(row.children.length).toBe(2);

    // test click triggers callback
    act(() => {
        row.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(onClick.mock.calls.length).toBe(1);
});