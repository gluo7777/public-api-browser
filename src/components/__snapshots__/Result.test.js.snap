// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result renders columns correctly 1`] = `
<li
  className="list-group-item"
>
  <div
    className="result h-pane"
    onClick={[Function]}
  >
    <p>
      <span>
        column1
      </span>
    </p>
    <p>
      <span>
        column2
      </span>
    </p>
  </div>
</li>
`;
